server/chat-plugins/repeats.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Chat plugin for repeating messages in chat
+ * Based on bot functionality from Kid A and Expecto Botronum
+ * @author Annika
+ */
+
+export interface RepeatedPhrase {
+	phrase: string;
+	/** interval in milliseconds */
+	interval: number;
+}
+
+export const pages: PageTable = {
+	repeats(args, user) {
+		const roomid = args[0];
+		this.title = `Repeated phrases in ${roomid}`;
+		const room = Rooms.get(roomid);
+		if (!room) throw new Chat.ErrorMessage(`<h2>${this.tr`No such room: "${roomid}".`}</h2>`);
+		this.checkCan("mute", null, room);
+		if (!room.settings.repeats) return `<h1>${this.tr`There are no repeated phrases in ${roomid}.`}</h1>`;
+		let html = `<div class="ladder pad"><h1>${this.tr`Repeated phrases in ${roomid}`}</h1>`;
+		html += `<table><tr><th>${this.tr`Phrase`}</th><th>${this.tr`Interval`}</th><th>${this.tr`Action`}</th>`;
+		html += `<button class="button" name="send" value="/join view-repeats-${roomid}" style="float: right">`;
+		html += `<i class="fa fa-refresh"></i> ${this.tr`Refresh`}</button>`;
+		for (const repeat of room.settings.repeats) {
+			const minutes = repeat.interval / (60 * 1000);
+			html += `<tr><td>${repeat.phrase}</td><td>${this.tr`every ${minutes} minute(s)`}</td>`;
+			html += `<td><button class="button" name="send" value="/removerepeat ${repeat.phrase}">${this.tr`Remove`}</button></td>`;
+		}
+		html += `</table></div>`;
+		return html;
+	},
+};
+
+export const commands: ChatCommands = {
+	repeat(target, room, user) {
+		room = this.requireRoom();
+		this.checkCan('mute', null, room);
+		const [intervalText, message] = this.splitOne(target);
```

Comment:
Why not call this intervalNum? intervalText is misleading for something that's going to be a number.

Reply:
I mean I understand. It's the text version to be parsed into a thing later.

Reply:
I wanted to indicate that it was a string (as opposed to `interval` which is a number). Maybe `intervalString`?

Reply:
to me, at least, it reads like the text you want to set in the interval. But, that's just me.
intervalString sounds better, I agree.